internal/provider/permissions/permission_resource.go

```
@@ -0,0 +1,276 @@
+func (r *PermissionResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state PermissionResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	perm, err := r.findPermissionByID(ctx, state.ID.ValueString())
+	if err != nil {
+		// Permission no longer exists → remove from state
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	diags := mapPermissionToModel(ctx, &state, perm)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+// Update is not supported
+func (r *PermissionResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddWarning(
+		"Update Not Supported",
+		"This resource is import-only. To change it, delete and re-import the correct permission.",
+	)
+	resp.Diagnostics.Append(req.State.Get(ctx, nil)...)
+}
+
+// Delete removes the permission via API
+func (r *PermissionResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state PermissionResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	permissionID, err := uuid.Parse(state.ID.ValueString())
+	tflog.Debug(ctx, "Deleting Entitle Permission", map[string]any{"id": permissionID})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("failed to parse account id to uuid format, got error: %s", err),
+		)
+		return
+	}
+	apiResp, err := r.client.PermissionsRevokeWithResponse(ctx, permissionID)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete permission %s: %s", permissionID, err))
+		return
+	}
+
+	if err := utils.HTTPResponseToError(apiResp.StatusCode(), apiResp.Body, utils.WithIgnorePending()); err != nil {
```

Worth splitting between errors
if 404 ->    Resource already deleted, this is fine for idempotency
All other use caeses should return an error